onnxscript/ir/_core.py

```diff
@@ -1676,7 +1681,7 @@ def Input(
 
 
 def _check_node_safe_to_remove(
-    node: Node, to_remove: AbstractSet[Node], graph_outputs: AbstractSet[Value]
+    node: Node, to_remove: AbstractSet[Node], graph_outputs: AbstractSet[int]
```

Comment:
I still think we should assume Values as hashable. This is required by the protocol (should make it explicit in text. cannot inherit Hashable because that requires explicit definition of `__hash__`.

Reply:
Yes, it would be convenient to use dictionaries with IR components as keys (or sets of them) ... if we can assume that, I will undo all related changes.

Reply:
Yes. Please assume that. I will fix any violations if you see any